src/main/java/org/spongepowered/common/item/enchantment/SpongeEnchantmentBuilder.java

```
@@ -0,0 +1,102 @@
+import org.spongepowered.common.data.util.DataUtil;
+
+import java.util.Optional;
+
+import javax.annotation.Nullable;
+
+public final class SpongeEnchantmentBuilder extends AbstractDataBuilder<Enchantment> implements Enchantment.Builder {
+
+    @Nullable private EnchantmentType enchantmentType;
+    @Nullable private Integer level;
+
+    public SpongeEnchantmentBuilder() {
+        super(Enchantment.class, 1);
+    }
+
+    @Override
+    public Enchantment.Builder from(final Enchantment value) {
+        this.enchantmentType = value.getType();
+        this.level = value.getLevel();
+        return this;
+    }
+
+    @Override
+    public Enchantment.Builder reset() {
+        this.enchantmentType = null;
+        this.level = null;
+        return this;
+    }
+
+    @Override
+    public Enchantment.Builder type(final EnchantmentType enchantmentType) {
+        this.enchantmentType = checkNotNull(enchantmentType, "Enchantment type cannot be null!");
+        return this;
+    }
+
+    @Override
+    public Enchantment.Builder level(final int level) {
+        checkArgument(level >= Short.MIN_VALUE, "The specified level must be greater than %s (was %s)!", Short.MIN_VALUE, level);
+        checkArgument(level <= Short.MAX_VALUE, "The specified level must not be greater than %s (was %s)!", Short.MAX_VALUE, level);
+        this.level = level;
+        return this;
+    }
+
+    @Override
+    public Enchantment build() {
+        checkState(this.enchantmentType != null, "The enchantment type must be set!");
+        checkState(this.level != null, "The level of the enchantment must be set!");
+        return new SpongeEnchantment(this.enchantmentType, this.level);
+    }
+
+    @Override
+    protected Optional<Enchantment> buildContent(DataView container) throws InvalidDataException {
+        checkNotNull(container, "The data view cannot be null!");
+        if (container.contains(Queries.ENCHANTMENT_ID, Queries.LEVEL)) {
```

Invert the if block, style change, I know, just preference. Fail fast.